internal/rbd/controllerserver.go

```diff
@@ -91,6 +91,36 @@ func (cs *ControllerServer) validateVolumeReq(ctx context.Context, req *csi.Crea
 	return nil
 }
 
+func getpercent(percent, size int64) int64 {
+	return ((size * percent) / 100)
+}
+
+func (cs *ControllerServer) validateThickOverProvisioning(vol *rbdVolume) error {
+	// LockName taken to validate the ceph pool size
+	poolSizeLock := "poolSizeLock"
+	if acquired := cs.VolumeLocks.TryAcquire(poolSizeLock); !acquired {
+		return status.Errorf(codes.Aborted, util.PoolOperationAlreadyExistsFmt, vol.Pool)
+	}
+	defer cs.VolumeLocks.Release(poolSizeLock)
+
+	cStats, err := getCephStats(vol.Monitors, *vol.conn.Creds)
+	if err != nil {
+		return status.Error(codes.Internal, err.Error())
+	}
+	for _, p := range cStats.Pools {
+		if p.Name == vol.Pool {
```

Comment:
Does this handle images with a separate data pool?

Reply:
Thanks for catching this one, added a check for the data pool also.